packages/data-broker/src/services/dhis/pullers/DeprecatedEventsPuller.ts

```diff
@@ -3,8 +3,18 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
+import type { DhisApi } from '@tupaia/dhis-api';
 import { EventsPuller } from './EventsPuller';
 
+type Options = Partial<{
```

Comment:
I've seen this Partial pattern used quite a bit in this PR. What's the advantage of Partial over the old school optional key syntax `{ a?: number }`. I like the old school way because it lets us do a mix of required and optional keys.

Reply:
I use `Partial` when all properties are de facto optional. This is a pattern we use often: the `options` object being entirely optional.

Happy to use the more explicit `?: ` syntax if you think it is clearer! Also tagging @rohan-bes for a second opinion

Reply:
Happy either way :)